newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -1,13 +1,77 @@
 package newrelic
 
 import (
-	"log"
-
+	"fmt"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"log"
+	"time"
 )
 
+func validateNaiveDateTime(val interface{}, key string) (warns []string, errs []error) {
+	valueString := val.(string)
+
+	// test conversion to desired format:
+	_, err := time.Parse("2006-01-02T15:04:05", valueString)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("%#v of %#v must be in the format 2006-01-02T15:04:05", key, valueString))
+	}
+	return
+}
+
+func scheduleSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"end_repeat": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The datetime stamp when the MutingRule schedule should stop repeating.",
+				ConflictsWith: []string{"schedule.repeat_count"},
+				ValidateFunc:  validateNaiveDateTime,
+			},
+			"end_time": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The datetime stamp representing when the MutingRule should end.",
+				ValidateFunc: validateNaiveDateTime,
+			},
+			"repeat": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The frequency the MutingRule schedule repeats. One of [DAILY, WEEKLY, MONTHLY]",
+				ValidateFunc: validation.StringInSlice([]string{"DAILY", "WEEKLY", "MONTHLY"}, false),
+			},
+			"repeat_count": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Description:   "The number of times the MutingRule schedule should repeat.",
+				ConflictsWith: []string{"schedule.end_repeat"},
+				ValidateFunc:  validation.IntAtLeast(1),
+			},
+			"start_time": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The datetime stamp representing when the MutingRule should start.",
+				ValidateFunc: validateNaiveDateTime,
+			},
+			"time_zone": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
Does this mean `time_zone` is required on create and update?  I'm trying to remember how we decided to handle it being optional on update in nerdgraph.

Reply:
I think it's transparently handled via the update. It's required when you first create a resource via terraform, and then it's only sent on update if it's been changed in the user's config (diffs vs tf state)

Reply:
Ok, so terraform won't throw a `time_zone is required` error if `time_zone` isn't provided in an update?

Reply:
Discussed this in pairing! TZ won't be sent on update if it hasn't changed in the TF config vs state (which is what we want)